pkg/helm/chartproxy/repos_test.go

```diff
@@ -512,3 +516,170 @@ func TestHelmRepoGetter_SkipDisabled(t *testing.T) {
+								"tlsClientConfig": map[string]interface{}{
+									"name":      "fooSecret",
+									"namespace": "testing",
+								},
+							},
+						},
+					},
+				}},
+			repoName:        "repo4",
+			wantsErr:        false,
+			createSecret:    true,
+			namespace:       "testing",
+			createNamespace: true,
+		},
+		{
+			name: "Namespace not present",
+			helmCRS: []*unstructured.Unstructured{
+				{
+					Object: map[string]interface{}{
+						"apiVersion": "helm.openshift.io/v1beta1",
+						"kind":       "ProjectHelmChartRepository",
+						"metadata": map[string]interface{}{
+							"namespace": "",
+							"name":      "repo5",
+						},
+						"spec": map[string]interface{}{
+							"connectionConfig": map[string]interface{}{
+								"url": "https://localhost:8080",
+								"tlsClientConfig": map[string]interface{}{
+									"name": "fooSecret",
+								},
+							},
+						},
+					},
+				}},
+			repoName:        "repo5",
+			wantsErr:        false,
+			createSecret:    true,
+			createNamespace: false,
+		},
+		{
+			name: "Namespace is invalid",
+			helmCRS: []*unstructured.Unstructured{
+				{
+					Object: map[string]interface{}{
+						"apiVersion": "helm.openshift.io/v1beta1",
+						"kind":       "ProjectHelmChartRepository",
+						"metadata": map[string]interface{}{
+							"namespace": "",
+							"name":      "repo6",
+						},
+						"spec": map[string]interface{}{
+							"connectionConfig": map[string]interface{}{
+								"url": "https://localhost:8080",
+								"tlsClientConfig": map[string]interface{}{
+									"name":      "fooSecret",
+									"namespace": 1,
+								},
+							},
+						},
+					},
+				}},
+			repoName:        "repo6",
+			wantsErr:        true,
+			createSecret:    true,
+			createNamespace: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			objs := []runtime.Object{}
+			// create a namespace if it is not same as openshift-config
+			if tt.createNamespace && tt.namespace != configNamespace {
+				nsSpec := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: tt.namespace}}
+				objs = append(objs, nsSpec)
+			}
+			// create a secret in required namespace
+			if tt.createSecret {
+				certificate, errCert := ioutil.ReadFile("./server.crt")
+				require.NoError(t, errCert)
+				key, errKey := ioutil.ReadFile("./server.key")
+				require.NoError(t, errKey)
+				data := map[string][]byte{
+					"tls.key": key,
+					"tls.crt": certificate,
+				}
+				if tt.namespace == "" {
+					tt.namespace = configNamespace
+				}
+				secretSpec := &v1.Secret{Data: data, ObjectMeta: metav1.ObjectMeta{Name: "fooSecret", Namespace: tt.namespace}}
+				objs = append(objs, secretSpec)
+			}
+			repoGetter := &helmRepoGetter{
+				Client:     fake.K8sDynamicClient("helm.openshift.io/v1beta1", "HelmChartRepository", ""),
+				CoreClient: k8sfake.NewSimpleClientset(objs...).CoreV1(),
+			}
+			_, err := repoGetter.unmarshallConfig(*tt.helmCRS[0])
```

Comment:
Only one repository is being used at a time in this test; could re-type `tt.helmCRS` from `[]*unstructured.Unstructured` to `*unstructured.Unstructured`.

Reply:
We are using the same function K8sDynamicClientFromCRs in all other test so that's why have declared these as a list of helmcrs despite being a single entity